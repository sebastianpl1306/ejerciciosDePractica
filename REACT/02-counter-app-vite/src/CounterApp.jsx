import { useState } from 'react';
import PropTypes from "prop-types";

export const CounterApp =  ({ value }) => {

    //El useState recibe como parametro el valor de inicialización y este se desestructura en una variable llamada counter y una función que permite modificar
    //el valor de counter
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>{
        // setCounter( counter + 1 );
        // setCounter(1000);
        setCounter((c)=> c + 1);
    }

    const handleSubstract = () => setCounter(counter-1);

    const handleReset = () => setCounter(value);

    return(
        <>
            <h1>counterApp</h1>
            <h2 data-testid="container-counter">{ counter }</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleSubstract}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.prototypes = {
    value: PropTypes.number
}