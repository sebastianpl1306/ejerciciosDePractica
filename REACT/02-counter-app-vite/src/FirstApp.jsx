import PropTypes from "prop-types";

const info = {
  name: 'Sebastian',
  lastname: 'Pabon Lopez'
};

const getName = (info) =>{
  return `Hello ${info.name} ${info.lastname} welcome!`;
}

export const FirstApp = ({title, subtitle, number}) => {

  return (
    <>
      <h1>{ getName(info) }</h1>
      <h2 data-testid='test-title'>{title}</h2>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>FirstApp {number+1}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number,
  subtitle: PropTypes.string
};

FirstApp.defaultProps = {
  number: 2
}