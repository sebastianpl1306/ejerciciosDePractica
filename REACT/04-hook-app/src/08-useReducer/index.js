export * from './TodoAdd';
export * from './TodoApp';
export * from './TodoItem';
export * from './TodoList';
export * from './todoReducer';