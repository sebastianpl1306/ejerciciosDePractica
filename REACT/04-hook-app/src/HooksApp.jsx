export const HooksApp = () => {
  return (
    <h1 className="text-primary">HooksApp</h1>
  )
}