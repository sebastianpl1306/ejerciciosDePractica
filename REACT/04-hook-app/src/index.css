body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

a{
    margin: 5px;
}