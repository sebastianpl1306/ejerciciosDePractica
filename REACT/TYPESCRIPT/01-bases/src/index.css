html, body{
  background-color: #282c34;
  color: white;
  margin: 0;
  padding: 10px;
}