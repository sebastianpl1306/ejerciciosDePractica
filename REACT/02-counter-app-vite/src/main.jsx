import React from 'react'
import ReactDOM from "react-dom/client";
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={0}></CounterApp>
        {/* <FirstApp title="Hola, soy Sebastiannn" number={123}></FirstApp> */}
    </React.StrictMode>
)