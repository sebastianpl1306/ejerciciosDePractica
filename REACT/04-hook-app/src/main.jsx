import React from 'react';
import ReactDOM from 'react-dom/client';
// import { Padre } from './07-tarea-memo/Padre';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { MultipleCustomHooks } from './03-examples/MultipleCustomHooks';
// import { FormWithCustomHook } from './02-useEfect/FormWithCustomHook';
// import { SimpleForm } from './02-useEfect/SimpleForm';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { CounterApp } from './01-useState/CounterApp';
// import { HooksApp } from './HooksApp';
// import './08-useReducer/intro-reducer';
import './index.css';
import { TodoApp } from './08-useReducer/TodoApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    <TodoApp />
  // </React.StrictMode>,
)
