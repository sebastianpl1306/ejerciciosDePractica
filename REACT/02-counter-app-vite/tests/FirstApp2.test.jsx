import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp/>', () => {
    const title = 'Hola, soy Sebastian';
    const subtitle = 'Soy un subtitulo';

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title}/>);
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola, soy Sebastian"', () => {
        render(<FirstApp title={ title }/>);
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe de mostrar el tÃ­tulo en el p', () => {
        render(<FirstApp title={ title }/>);
        expect( screen.getByRole('heading', {level: 2}).innerHTML ).toContain(title);
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle}/>);
        expect( screen.getAllByText(subtitle).length ).toBe(2);
    })
})