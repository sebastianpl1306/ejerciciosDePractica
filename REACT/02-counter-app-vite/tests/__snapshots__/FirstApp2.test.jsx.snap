// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hello Sebastian Pabon Lopez welcome!
  </h1>
  <h2
    data-testid="test-title"
  >
    Hola, soy Sebastian
  </h2>
  <p />
  <p />
  <p>
    FirstApp 
    3
  </p>
</div>
`;
